main.go

```diff
@@ -82,6 +108,155 @@ func runMain() error {
 		fmt.Println("The local master branch is not in sync with the remote master branch. Must be in sync to conduct release process.")
 		return nil
 	}
+	
+	// Guess the next release version
+	latestReleaseVersion := getLatestReleaseVersion(repository, NO_PREVIOUS_VERSION)
+
+	// Conduct changelog file validation
+	changelogFilepath := path.Join(currentWorkingDirectory, relChangelogFilepath)
+
+	tbdHeaderCount := grepFile(changelogFilepath, TBD_VERSION_HEADER_REGEX)
+	if tbdHeaderCount != EXPECTED_NUM_TBD_HEADER_LINES {
+		fmt.Printf("There should be %d TBD header lines in the changelog. Instead there are %d.\n", EXPECTED_NUM_TBD_HEADER_LINES, tbdHeaderCount)
+		return nil
+	}
+
+	versionHeaderCount := grepFile(changelogFilepath, VERSION_HEADER_REGEX)
+	if versionHeaderCount == 0 {
+		fmt.Println("No previous changelog versions were detected in this changelog. Are you sure that the changelog is in sync with the release tags on this branch?")
+		return nil
+	}
+
+	existsBreakingChanges := detectBreakingChanges(changelogFilepath)
```

Comment:
to make the name flow smoother, I'll often do something like `doBreakingChangesExist` (so that we fulfill the boolean func/var naming contract)